### DISCUSSION TITLE
Budibase app as standalone frontend?

### DISCUSSION THREAD

**Discussion by itsmarkvm** [Budibase/budibase#7455] [Help]
Is it possible to use Budibase to develop a lightweight front-end only solution? So without any backend parts?

I'm currently in the process of building a UI for existing GraphQL end points. The UI will only be used by a handful of people so we don't need any customization or branding, we just need a UI that will enable us to query and edit data through the GraphQL endpoint. Budibase seems like an excellent tool for me to quickly build something very usable.

My stack is, as much as possible, based around serverless AWS. For control and security reasons I would like to control the hosting myself. That's why I'm initially not interested in the hosted Budibase solution.

Based on the documentation I get the impression if I want to use Budibase for this use case I need to install the entire Budibase suite, including the dev environment, in a Docker container on AWS. This is overkill for this particular use case as I only need the HTML, JS and CSS parts of the generated application.

I did build a small app that only interfaces with an external API, but I noticed the app still made some roundtrips to the Budibase servers. 

Does anybody know if it is possible to separate the produced application from the Budibase suite and deploy it on S3 or Nginx or so?

**Comment by melohagan** [comment]
Hi @itsmarkvm 

I've seen this asked once before. I think the bottom line is that all the moving parts are critical and can't be separated. As you noticed, the apps will actively communicate with the server container. 
Querying and filtering the data for example is not handled client side. 

You may get better performance from the docker single image: https://docs.budibase.com/docs/docker#method-1---docker-run
